api/envoy/extensions/filters/http/oauth/v3/oauth.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.oauth.v3;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.oauth.v3";
+option java_outer_classname = "OauthProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: OAuth]
+// OAuth :ref:`configuration overview <config_http_filters_oauth>`.
+// [#extension: envoy.filters.http.oauth]
+//
+
+message OAuth2Credentials {
+  // The client_id to be used in the authorize calls.
+  string client_id = 1;
+
+  // The config name in SDS that contains the client secret that will be used
+  // to get an access token.
+  string client_secret_config_name = 2;
+
+  // The config name in SDS that contains the token secret that will be used
+  // to perform the HMAC validation.
+  string token_secret_config_name = 3;
+
+  // The cluster that has the SDS generic secrets.
+  string secrets_cluster = 4 [(validate.rules).string = {min_bytes: 1}];
```

Comment:
Why isn't this a generic `ConfigSource`? The implementation seems to tie this to Envoy gRPC (so Google gRPC would not be usable).

Reply:
I don't think there's a good reason, let me change it